src/libraries/Microsoft.Extensions.Logging.Abstractions/gen/LoggerMessageGenerator.Parser.cs

```diff
@@ -601,6 +601,27 @@ public IReadOnlyList<LoggerClass> GetLogClasses(IEnumerable<ClassDeclarationSynt
                 return results;
             }
 
+            private static string GenerateClassName(TypeDeclarationSyntax typeDeclaration)
+            {
+                if (typeDeclaration.TypeParameterList != null &&
+                    typeDeclaration.TypeParameterList.Parameters.Count != 0)
+                {
+                    // Strip any type parameter attributes, otherwise compilation will fail due to
+                    // duplicate declarations. See https://github.com/dotnet/runtime/issues/97498.
+                    for (int i = 0; i < typeDeclaration.TypeParameterList.Parameters.Count; i++)
+                    {
+                        TypeParameterSyntax parameter = typeDeclaration.TypeParameterList.Parameters[i];
+
+                        if (parameter.AttributeLists.Count > 0)
+                        {
+                            typeDeclaration = typeDeclaration.ReplaceNode(parameter, parameter.WithAttributeLists([]));
```

Comment:
>typeDeclaration = typeDeclaration.ReplaceNode(parameter, parameter.WithAttributeLists([])); [](http://example.com/codeflow?start=28&length=91)

Have you verified whether the final IL code generated by the compiler still contains the attribute? While we are addressing the issue in the generated code, it is crucial to confirm that this correction does not unintentionally remove the attribute from the combined partial classes emitted by the compiler.

Reply:
I hadn't as it didn't occur to me that this could be possible as my understanding is that it's editing the model which is then compiled for the source emitted, rather than changing the total sum of the source for the target type from all its sources.

I've just checked, and the attributes are still present if I view the compiled test DLL in ILSpy:

![image](https://github.com/dotnet/runtime/assets/1439341/826034f2-4a07-4a36-bfdf-0c8fe3e135b6)